packages/babel-traverse/src/path/inference/inferers.js

```diff
@@ -137,7 +137,13 @@ function Func() {
   return t.genericTypeAnnotation(t.identifier("Function"));
 }
 
-export { Func as Function, Func as Class };
+export {
+  Func as FunctionExpression,
```

Comment:
I interpreted it to be an implementation detail, where `getTypeAnnotation()`, `baseTypeStrictlyMatches()` etc are the actual API.